### DISCUSSION TITLE
Error on command:./run-docker.sh

### DISCUSSION THREAD

**Discussion by lloo099** [Xilinx/finn#429] [Q&A] [answered]
Hi, I encounter one error about running finn
![Screenshot from 2021-10-24 17-50-09](https://user-images.githubusercontent.com/29093207/138588974-a70da622-e5d9-488a-b175-0373f51a67b5.png)

**Comment by lloo099** [comment]
Does anyone knows this?

**Comment by fpjentzsch** [comment]
Hi,
which FINN branch/version are you using and how did you clone and run the docker container?

**Comment by lloo099** [comment]
> Hi, which FINN branch/version are you using and how did you clone and run the docker container?

Hi, I used the FINN version which is downloaded one month ago, and not sure which one. I just downloaded it as .zip file from github and unzip on my computer. Then I run ./run-docker.sh build_custom /home/enai/Desktop/project/finn/build/fpga_flow

**Comment by lloo099** [comment]
Successfully installed PyVerilator tclwrapper-0.0.1
oh-my-xilinx @ 1bae737669901e762f581af73348332b5c4b2ada
Obtaining file:///workspace/finn
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... error
    ERROR: Command errored out with exit status 1:
     command: /opt/conda/bin/python /opt/conda/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /tmp/tmpmn67wo0i
         cwd: /workspace/finn
    Complete output (19 lines):
    running dist_info
    creating /tmp/pip-modern-metadata-c5_ci_fk/finn_examples.egg-info
    writing /tmp/pip-modern-metadata-c5_ci_fk/finn_examples.egg-info/PKG-INFO
    writing dependency_links to /tmp/pip-modern-metadata-c5_ci_fk/finn_examples.egg-info/dependency_links.txt
    writing entry points to /tmp/pip-modern-metadata-c5_ci_fk/finn_examples.egg-info/entry_points.txt
    writing requirements to /tmp/pip-modern-metadata-c5_ci_fk/finn_examples.egg-info/requires.txt
    writing top-level names to /tmp/pip-modern-metadata-c5_ci_fk/finn_examples.egg-info/top_level.txt
    writing manifest file '/tmp/pip-modern-metadata-c5_ci_fk/finn_examples.egg-info/SOURCES.txt'
    /tmp/pip-build-env-bfec9xq5/overlay/lib/python3.8/site-packages/setuptools/dist.py:487: UserWarning: Normalizing '0.0.3b' to '0.0.3b0'
      warnings.warn(tmpl.format(**locals()))
    /tmp/pip-build-env-bfec9xq5/overlay/lib/python3.8/site-packages/setuptools/dist.py:717: UserWarning: Usage of dash-separated 'author-email' will not be supported in future versions. Please use the underscore name 'author_email' instead
      warnings.warn(
    /tmp/pip-build-env-bfec9xq5/overlay/lib/python3.8/site-packages/setuptools/dist.py:717: UserWarning: Usage of dash-separated 'long-description' will not be supported in future versions. Please use the underscore name 'long_description' instead
      warnings.warn(
    /tmp/pip-build-env-bfec9xq5/overlay/lib/python3.8/site-packages/setuptools/dist.py:717: UserWarning: Usage of dash-separated 'long-description-content-type' will not be supported in future versions. Please use the underscore name 'long_description_content_type' instead
      warnings.warn(
    /tmp/pip-build-env-bfec9xq5/overlay/lib/python3.8/site-packages/setuptools/dist.py:717: UserWarning: Usage of dash-separated 'project-urls' will not be supported in future versions. Please use the underscore name 'project_urls' instead
      warnings.warn(
    error: package directory 'src/finn_examples' does not exist
    ----------------------------------------
ERROR: Command errored out with exit status 1: /opt/conda/bin/python /opt/conda/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /tmp/tmpmn67wo0i Check the logs for full command output.
/tools/Xilinx/Vivado/2020.1/settings64.sh: line 5: /tools/Xilinx/DocNav/.settings64-DocNav.sh: No such file or directory
/usr/local/bin/finn_entrypoint.sh: line 103: /tools/Xilinx/Vitis/settings64.sh: No such file or directory
XILINX_XRT      : /opt/xilinx/xrt
PATH            : /opt/xilinx/xrt/bin:/tools/Xilinx/Vitis/2020.1/bin:/tools/Xilinx/Vitis/2020.1/gnu/microblaze/lin/bin:/tools/Xilinx/Vitis/2020.1/gnu/arm/lin/bin:/tools/Xilinx/Vitis/2020.1/gnu/microblaze/linux_toolchain/lin64_le/bin:/tools/Xilinx/Vitis/2020.1/gnu/aarch32/lin/gcc-arm-linux-gnueabi/bin:/tools/Xilinx/Vitis/2020.1/gnu/aarch32/lin/gcc-arm-none-eabi/bin:/tools/Xilinx/Vitis/2020.1/gnu/aarch64/lin/aarch64-linux/bin:/tools/Xilinx/Vitis/2020.1/gnu/aarch64/lin/aarch64-none/bin:/tools/Xilinx/Vitis/2020.1/gnu/armr5/lin/gcc-arm-none-eabi/bin:/tools/Xilinx/Vitis/2020.1/tps/lnx64/cmake-3.3.2/bin:/tools/Xilinx/Vitis/2020.1/cardano/bin:/tools/Xilinx/Vivado/2020.1/bin:/opt/conda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/workspace/oh-my-xilinx:/home/enai/.local/bin
LD_LIBRARY_PATH : /opt/xilinx/xrt/lib:/usr/local/nvidia/lib:/usr/local/nvidia/lib64
PYTHONPATH     : /opt/xilinx/xrt/python:
Traceback (most recent call last):
  File "/opt/conda/lib/python3.8/pdb.py", line 1704, in main
    pdb._runscript(mainpyfile)
  File "/opt/conda/lib/python3.8/pdb.py", line 1573, in _runscript
    self.run(statement)
  File "/opt/conda/lib/python3.8/bdb.py", line 580, in run
    exec(cmd, globals, locals)
  File "<string>", line 1, in <module>
  File "/workspace/Desktop/project/finn/build/fpga_flow/build.py", line 36, in <module>
    import finn.builder.build_dataflow as build
ModuleNotFoundError: No module named 'finn.builder'
Uncaught exception. Entering post mortem debugging
Running 'cont' or 'step' will restart the program
Post mortem debugger finished. The build.py will be restarted
> /workspace/Desktop/project/finn/build/fpga_flow/build.py(36)<module>()
-> import finn.builder.build_dataflow as build
(Pdb) exit

I run it again, and it shows like these

**Comment by fpjentzsch** [comment, marked as answer]
It can cause trouble if you simply download the repository as a zip file. Can you try to properly clone the repo and checkout the current dev branch?

**Comment by lloo099** [comment]
> It can cause trouble if you simply download the repository as a zip file. Can you try to properly clone the repo and checkout the current dev branch?
Thanks, I try it again from clone. But the program creaked  because vivado_hls path. I already export VIVADO_PATH and VITIS_PATH before run docker.sh. Could you help me to check my codes and some suggestions? 
export VIVADO_PATH=/tools/Xilinx/Vivado/2020.1
export VITIS_PATH=/tools/Xilinx/Vitis

/home/enai/Pictures/Screenshot from 2021-11-16 20-55-19.png
![Screenshot from 2021-11-16 20-55-19](https://user-images.githubusercontent.com/29093207/141989424-c0a94fe8-0c24-480e-bd8e-c241af6f4ee7.png)

**Comment by fpjentzsch** [comment]
This was changed recently: Instead of `VIVADO_PATH` and `VITIS_PATH` you now have to set `FINN_XILINX_PATH`and `FINN_XILINX_VERSION`

For example:
`export FINN_XILINX_PATH=/tools/Xilinx`
`export FINN_XILINX_VERSION=2020.1`

**Comment by Ba1tu3han** [comment]
I encountered the same problem, too. I have cloned 0.9.2 version of FINN depending on the getting started web page. And I run it by "./run-docker.sh quicktest" command. 

I have removed and cleaned everything and done it again. It is not fixed. 

What do you recommend me ? @fpjentzsch 

Thank you,

![image](https://user-images.githubusercontent.com/29502318/221409535-e4039848-4e91-4e44-9408-d48c94ed68b6.png)
